pygam/pygam.py

```diff
@@ -1781,6 +1781,59 @@ def gridsearch(self, X, y, return_scores=False, keep_best=True,
         else:
             return self
 
+    def simulate_from_coef_posterior_conditioned_on_smoothing_parameters(
+            self, X, n_simulations=100):
+        """Simulate from the posterior of distribution over the coefficients a
+        GAM, conditioned on its smoothing parameters.
+
+        A number of random samples of the coefficients are drawn from a
+        multivariate normal distribution with mean vector given by the
+        coefficients `self.coef_` and covariance given by
+        `self.statistics_['cov']`.
+
+        Parameters
+        ---------
+        X : array-like of shape (n_samples, m_features)
+            The inputs to the model
+
+        n_simulations : int, default: 100
+            The number of random samples to draw from the posterior of the
+            distribution over the coefficients
+
+        Returns
+        -------
+        simulated_mu : np.array of shape (n_simulations, n_samples)
+            Random samples of the expected value, mu, of the target, with
+            coefficients of the model drawn from its posterior distribution
+            (conditioned on the smoothing parameters being equal to the values
+            already estimated).
+
+        References
+        ----------
+        Simon N. Wood, 2006. Generalized Additive Models: an introduction with
+        R. Section 5.2.7 (pages 242–243) and Section 5.4.2 (page 256).
+        """
+        if not self._is_fitted:
+            raise AttributeError('GAM has not been fitted. Call fit first.')
+
+        if n_simulations <= 0:
```

Comment:
👍 
this should be good enough.
I dont feel like checking if its an `int` etc. numpy will check for us.